docs/content/chapter_3.md

```diff
@@ -0,0 +1,85 @@
+# Project Environment
+
+## Introduction
+
+In here we'll discuss about the project environment. We'll discuss about the project architecture & folder structure, project environments (`Production`, `Development`, `Staging`), how secrets `(environment variables)` are handled, how to use environment variables for different environments,
+
+### Project Architecture & Folder Structure
+
+The project architecture is based on the [Next.js](https://nextjs.org/) framework. The project folder structure is as follows.
+
+```bash
+├── .github
+│   └── workflows
+│       └── ci.yml
+├── .next
+│   ├── cache
+│   ├── dist
+│   ├── server
+│   └── static
+├── .vscode
+│   └── settings.json
+├── components
+│   ├── common
+│   │   ├── Button
+│   │   │   ├── index.js
+│   │   │   └── styles.js
+│   │   ├── Card
+│   │   │   ├── index.js
+│   │   │   └── styles.js
+│   │   ├── Footer
+│   │   │   ├── index.js
+│   │   │   └── styles.js
+│   │   ├── Header
+│   │   │   ├── index.js
+│   │   │   └── styles.js
+│   │   ├── Layout
+│   │   │   ├── index.js
+│   │   │   └── styles.js
+│   │   └── Navbar
+│   │       ├── index.js
+│   │       └── styles.js
+│   └── pages
+│       ├── blog
+│       │   ├── [slug].js
+│       │   └── index.js
+│       ├── index.js
+│       └── post
+│           └── [...all].js
+├── next.config.js
+├── package.json
+├── pages
+│   ├── _app.js
+│   ├── _document.js
+│   ├── _error.js
+│   ├── blog
+│   │   ├── [slug].js
+│   │   └── index.js
+│   ├── index.js
+│   └── post
+│       └── [...all].js
+├── public
+│   ├── favicon.ico
+│   └── images
+│       └── logo.png
+├── README.md
+├── styles
+│   ├── global.js
+│   └── theme.js
+└── yarn.lock
+```
+
+### Project Environments
+
+The project has three environments. They are `Production`, `Development`, `Staging`. For each type of environment, we have a separate `.env` file. [Read more about environment variables](#environment-variables--how-we-use-them).
```

Comment:
Better to have an unordered list with a short explanation.

**The project has three environments.** 

- **Production (Host the live version of the project and deploy the final release to the public)** 
- **Development (Develop and test new features and changes before they are released to Staging and Production)** 
- **Staging (Test the project with real data)** 

**For each type of environment, we have a separate `.env` file because it allows us to control what environment the project is running in and provide any environment-specific variables such as passwords, credentials, or API keys.**

**We use different databases for each environment since it helps to guarantee that production data is used only in production and staging data is used only in staging.**

**We also use different servers for each environment, so the production server is separate from the staging server. It helps to protect that production code is only deployed to production and staging code is only deployed to staging.**

Reply:
done